### ISSUE TITLE
Multihandler test failure

### ISSUE THREAD

**Issue by zzzanderw** [CS362-Group/EggShell#13]
```
ERROR: test_background_server (test_multihandler.TestMultiHandler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/zander/EggShell/test/test_multihandler.py", line 29, in test_background_server
    multihandler_obj.stop_server()
  File "/home/zander/EggShell/modules/multihandler.py", line 104, in stop_server
    socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect((self.server.host,self.server.port))
  File "/usr/lib/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
TypeError: coercing to Unicode: need string or buffer, NoneType found
```
@okonekp is this expected?

**Comment by zzzanderw** [comment]
Also, my test client hangs after outputting:
```
----------------------------------------------------------------------
Ran 19 tests in 3.786s

FAILED (errors=1)

```
This is related to that test, b/c I commented it out and the test suite no longer hangs

**Comment by zzzanderw** [comment]
Forgot to mention that Ctrl-C doesn't kill the program

**Comment by zzzanderw** [comment]
@okonekp oof
```
ERROR: test_background_server (test_multihandler.TestMultiHandler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/zander/EggShell/test/test_multihandler.py", line 27, in test_background_server
    server_obh.host = "dummy"
NameError: global name 'server_obh' is not defined
```

**Comment by zzzanderw** [comment]
I fixed it to server_obj, getting a different error:
```
ERROR: test_background_server (test_multihandler.TestMultiHandler)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/zander/EggShell/test/test_multihandler.py", line 31, in test_background_server
    multihandler_obj.stop_server()
  File "/home/zander/EggShell/modules/multihandler.py", line 104, in stop_server
    socket.socket(socket.AF_INET, socket.SOCK_STREAM).connect((self.server.host,self.server.port))
  File "/usr/lib/python2.7/socket.py", line 228, in meth
    return getattr(self._sock,name)(*args)
gaierror: [Errno -2] Name or service not known

----------------------------------------------------------------------
```

**Comment by zzzanderw** [comment]
This was fixed in a0630dcf23e599474018293245f6c80c85c0f97a
